File: dart/avoid_flag_arguments.dart
// Avoid Flag Arguments
// Avoid providing boolean arguments (true/false) to functions or methods.
// You could pass a string with a clearer name (for example).

// Before
void greetBefore(String name, bool formal) {
  if (formal) {
    print("Good evening, $name.");
  } else {
    print("Hi, $name!");
  }
}

// After
void greet(String name, String formality) {
  switch (formality) {
    case "formal":
      print("Good evening, $name.");
      break;
    case "informal":
      print("Hi, $name!");
      break;
  }
}

void main() {
  greetBefore("Alice", true);
  greetBefore("Bob", false);

  print("");

  greet("Alice", "formal");
  greet("Bob", "informal");
}

File: java/AvoidFlagArguments.java
// Avoid Flag Arguments
// Avoid providing boolean arguments (true/false) to functions or methods.
// You could pass a string with a clearer name (for example).

public class AvoidFlagArguments {
    // Before
    public static void greetBefore(String name, boolean formal) {
        if (formal) {
            System.out.println("Good evening, " + name + ".");
        } else {
            System.out.println("Hi, " + name + "!");
        }
    }

    // After
    public static void greet(String name, String formality) {
        if ("formal".equals(formality)) {
            System.out.println("Good evening, " + name + ".");
        } else if ("informal".equals(formality)) {
            System.out.println("Hi, " + name + "!");
        }
    }

    public static void main(String[] args) {
        greetBefore("Alice", true);
        greetBefore("Bob", false);

        System.out.println();

        greet("Alice", "formal");
        greet("Bob", "informal");
    }
}
